### PR TITLE
Documentation: Update spatial indexing example

### PR DESCRIPTION
**Pull request by ektravel**
Fixes https://github.com/apache/druid/issues/9970.

Description
This PR fixes the issue reported onhttps://github.com/apache/druid/issues/9970.

This PR has:

[ x ] been self-reviewed.

### CODE REVIEW

File: docs/development/geo.md [hunk 1]
```diff
@@ -23,13 +23,14 @@ title: "Spatial filters"
   -->
 
 > Apache Druid supports two query languages: [Druid SQL](../querying/sql.md) and [native queries](../querying/querying.md).
-> This document describes functionality that is only available in the native language.
+> This document describes the feature that is only available in the native language.
```
Comment (vtlim) [inline docs/development/geo.md:26]:
```suggestion
> This document describes a feature that is only available in the native language.
```

File: docs/development/geo.md [hunk 2]
```diff
@@ -23,13 +23,14 @@ title: "Spatial filters"
...
 
 Apache Druid supports filtering specially spatially indexed columns based on an origin and a bound.
 
 ## Spatial indexing
 
-In any of the data specs, there is the option of providing spatial dimensions. For example, for a JSON data spec, spatial dimensions can be specified as follows:
+In any of the data specs, there is the option of providing spatial dimensions.
```
Comment (vtlim) [inline docs/development/geo.md:32]:
```suggestion
You can provide spatial dimensions in any data spec.
```

**Review by vtlim — APPROVED** [review]
Left a couple of suggestions, otherwise LGTM

**Comment by vtlim** [comment]
@ektravel -- possibly worth including the comment raised in the linked issue (for the description of `dims`):
>  The order of the dimensions used in the spatial ingestion spec must be equal to the order of the dimensions used in the filtering spec.

**Comment by ektravel** [comment]
@vtlim This PR is ready for review. It includes expanded content and updated examples. Technical review might be required to ensure complete accuracy.

File: docs/development/geo.md [hunk 3]
```diff
@@ -23,53 +23,83 @@ title: "Spatial filters"
...
 
-|property|description|required?|
-|--------|-----------|---------|
-|dimName|The name of the spatial dimension. A spatial dimension may be constructed from multiple other dimensions or it may already exist as part of an event. If a spatial dimension already exists, it must be an array of coordinate values.|yes|
-|dims|A list of dimension names that comprise a spatial dimension.|no|
+A filter is a JSON object indicating which rows of data should be included in the computation for a query.
+You can filter on spatial structures, such as rectangles and polygons, using the spatial filter.
 
-The grammar for a spatial filter is as follows:
+The spatial filter has the following structure:
```
Comment (vtlim) [inline docs/development/geo.md:97]: This is a specific example and not a generic structure. We could add something like the following:

```
"filter": {
  "type": "spatial",
  "dimension": <name_of_spatial_dimension>,
  "bound": <bound_type>
}
```

That way we can have the generic grammar, then introduce a specific example.

File: docs/development/geo.md [hunk 4]
```diff
@@ -23,53 +23,83 @@ title: "Spatial filters"
...
+
+This topic explains how to apply spatial filters.
+For information on other filters supported by Druid, see [Query filters](../querying/filters.md).
 
 ## Spatial indexing
 
-In any of the data specs, there is the option of providing spatial dimensions. For example, for a JSON data spec, spatial dimensions can be specified as follows:
+Spatial indexing refers to ingesting data of a spatial data type, such as geometry or geography, into Druid.
+
+Spatial dimensions are sting columns that contain coordinates separated by a comma.
```
[the remark this answers is not in the document]
Comment (vtlim) [reply docs/development/geo.md:37]: I think this describes the result of ingestion a spatial dimension. Consider starting with how a spatial dimension gets ingested? Possibly something like, "Spatial dimensions are derived (generated? composed?) from two or more specified input dimensions. Druid stores spatial dimensions as comma-separated values in string columns."

File: docs/development/geo.md [hunk 5]
```diff
@@ -23,53 +23,83 @@ title: "Spatial filters"
...
+|`dimName`|The name of a spatial dimension. You can construct a spatial dimenion from other dimensions or it may already exist as part of an event. If a spatial dimension already exists, it must be an array of coordinate values.|yes|
+|`dims`|The list of dimension names that comprise the spatial dimension.|no|
+
 ## Spatial filters
 
-|property|description|required?|
-|--------|-----------|---------|
-|dimName|The name of the spatial dimension. A spatial dimension may be constructed from multiple other dimensions or it may already exist as part of an event. If a spatial dimension already exists, it must be an array of coordinate values.|yes|
-|dims|A list of dimension names that comprise a spatial dimension.|no|
+A filter is a JSON object indicating which rows of data should be included in the computation for a query.
```
Comment (vtlim) [inline docs/development/geo.md:94]:
```suggestion
A [filter](../querying/filters.md) is a JSON object indicating which rows of data should be included in the computation for a query.
```

File: docs/development/geo.md [hunk 6]
```diff
@@ -23,53 +23,83 @@ title: "Spatial filters"
...
 
 ## Spatial indexing
 
-In any of the data specs, there is the option of providing spatial dimensions. For example, for a JSON data spec, spatial dimensions can be specified as follows:
+Spatial indexing refers to ingesting data of a spatial data type, such as geometry or geography, into Druid.
+
+Spatial dimensions are sting columns that contain coordinates separated by a comma.
+In the ingestion spec, you configure spatial dimensions in the `dimensionsSpec` object of the `dataSchema` component.
+For information on how to use the ingestion spec to configure ingestion, see [Ingestion spec reference](../ingestion/ingestion-spec.md).
+For general information on loading data in Druid, see [Ingestion](../ingestion/index.md).
```
Comment (vtlim) [inline docs/development/geo.md:40]: Maybe move these lines to the end of the section, before spatial filters?

File: docs/development/geo.md [hunk 7]
```diff
@@ -78,25 +108,36 @@ The grammar for a spatial filter is as follows:
...
+The `polygon` bound has the following elements:
 
-|property|description|required?|
-|--------|-----------|---------|
-|abscissa|Horizontal coordinate for corners of the polygon|yes|
-|ordinate|Vertical coordinate for corners of the polygon|yes|
+|Property|Description|Required|
+|--------|-----------|--------|
+|`abscissa`|Horizontal coordinate for the corners of the polygon|yes|
+|`ordinate`|Vertical coordinate for the corners of the polygon|yes|
```
Comment (vtlim) [inline docs/development/geo.md:143]:
```suggestion
|`abscissa`|Horizontal coordinates for the corners of the polygon|yes|
|`ordinate`|Vertical coordinates for the corners of the polygon|yes|
```
https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/collections/spatial/search/PolygonBound.java#L82-L85

File: docs/development/geo.md [hunk 8]
```diff
@@ -78,25 +108,36 @@ The grammar for a spatial filter is as follows:
...
+You can define rectangular, radius, and polygon filter bounds.
+
+#### Rectangular
+
+The `rectangular` bound has the following elements:
+
+|Property|Description|Required|
+|--------|-----------|--------|
+|`minCoords`|The list of minimum dimension coordinates in the form [latitude, longitude]|yes|
+|`maxCoords`|The list of maximum dimension coordinates in the form [latitude, longitude]|yes|
```
Comment (vtlim) [inline docs/development/geo.md:125]:
```suggestion
|`minCoords`|The list of minimum dimension coordinates in the form [x, y]|yes|
|`maxCoords`|The list of maximum dimension coordinates in the form [x, y]|yes|
```

**Review by vtlim — COMMENTED** [review]
Nice additions! Left some suggestions.

**Comment by FrankChen021** [comment]
This PR stays for a long time. I just committed all the suggestions that @vtlim submitted because these suggestions looks good to me.
And there're still two comments left for @ektravel for final check.

**Comment by FrankChen021** [comment]
Thank you @ektravel and @vtlim

**Pull request merged**